File: calculator.py

class Calculator:
    def add(self, a, b):
        return a + b

    def subtract(self, a, b):
        return a - b

    def multiply(self, a, b):
        return a * b

    def divide(self, a, b):
        if b == 0:
            raise ValueError("Cannot divide by zero")

        return a / b

#1
if __name__ == "__main__":
    calc = Calculator()
    print("Addition (3 + 5):", calc.add(3, 5))
    print("Subtraction (10 - 4):", calc.subtract(10, 4))
    print("Multiplication (6 * 7):", calc.multiply(6, 7))
    print("Division (8 / 2):", calc.divide(8, 2))
    try:
        calc.divide(5, 0)
    except ValueError as e:
        print("Division by zero error:", e)

File: woss/src/Calculator.java
public class Calculator {

    // 函数
    public double add(double a, double b) {
        return a + b;
    }

    // 减法
    public double subtract(double a, double b) {
        return a - b;
    }

    // 乘法
    public double multiply(double a, double b) {
        return a * b;
    }

    // 除法
    public double divide(double a, double b) {
        if (b == 0) {
            throw new IllegalArgumentException("除数不能为零");
        }
        return a / b;
    }

    public static void main(String[] args) {
        Calculator calculator = new Calculator();

        double a = 10.0;
        double b = 5.0;

        System.out.println("加法: " + calculator.add(a, b));
        System.out.println("减法: " + calculator.subtract(a, b));
        System.out.println("乘法: " + calculator.multiply(a, b));
        System.out.println("除法: " + calculator.divide(a, b));
    }
}
